src/utils/daemon.js

```diff
@@ -1,42 +1,109 @@
 import IPFSFactory from 'ipfsd-ctl'
 import logger from './logger'
+import { showRepoApiFileErrorMessage } from './errors'
+import { join } from 'path'
+import fs from 'fs-extra'
+import { spawnSync } from 'child_process'
+import findExecutable from 'ipfsd-ctl/src/utils/find-ipfs-executable'
 
-export default async function createDaemon (opts) {
-  opts.type = opts.type || 'go'
-  opts.path = opts.path || ''
-  opts.flags = opts.flags || ['--migrate=true', '--routing=dhtclient']
-  opts.keysize = opts.keysize || 4096
+function repoFsck (path) {
+  const exec = findExecutable('go', join(__dirname, '..'))
+  spawnSync(exec, ['repo', 'fsck'], {
+    env: {
+      ...process.env,
+      IPFS_PATH: path
+    }
+  })
+}
+
+async function cleanup (path) {
+  const apiFile = join(path, 'api')
+
+  if (!await fs.pathExists(apiFile)) {
+    return true
```

Comment:
Let's make the api for `cleanup` clearer. Having `cleanup` return true if it doens nothing is suprising. I think we need more info than just a boolean return value.

There are few other issues
- `const cfg = await fs.readJSON(cfgFile)` could throw if the config isn't valid json. This function should be robust and try it's best to detect what's going on.
- `if (addr === cfgAddr && (lockFile || datastoreLock)) {` I think the situatuon we're looking for is just "there is an api file and no lockFile and no datastroreLock.", then it's safe to just run fsck. If there is a lockfile and an apiFile, but we failed to connect to the api, it'd be ok to ask the user if we should try and fsck. We shouldn't offer to fsck the repo if there is alredy a daemon running that we just can't connect to for some reason.

Let's pull this function out to a helper `detectOrphanApiFile` that returns true if the api File is present and the same as the config default value, and that there is no repo.lock and no datastore/LOCK.

Reply:
> - `api` and `repo.lock` are present but no daemon running AND  `api` points at the same multiaddr as `config`  

@lidel is it possible to assert that no daemon is running? is an  `ECONNREFUSED` from the api enough?

Reply:
I'd say `ECONNREFUSED` its the best platform-agnostic hint we can get.

Alternative is to execute platform-specific process enumeration(`tasklist` on windows, `ps -A` on linux, `ps something` on mac) to see if process is running, which may get blocked and not work reliably in more locked down systems.